### PR TITLE
fix: type-hinting in ScheduledEvent.subscribers

### PR DESCRIPTION
**Pull request by yoggys**
## Summary

Fixed type-hinting for `ScheduledEvent.subscribers` parameter `limit: Optional[:class:int]`

## Information

- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [ ] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

**Comment by yoggys** [comment]
ahaha wrong title

File: discord/iterators.py [hunk 1]
```diff
@@ -852,7 +852,7 @@ class ScheduledEventSubscribersIterator(_AsyncIterator[Union["User", "Member"]])
     def __init__(
         self,
         event: ScheduledEvent,
-        limit: int,
+        limit: int | None,
         with_member: bool = False,
         before: datetime.datetime | int = None,
         after: datetime.datetime | int = None,
```
Comment (plun1331) [inline discord/iterators.py:858]: We could add `| None` to this too

Comment (yoggys) [reply discord/iterators.py]: > We could add `| None` to this too

yeah

**Review by plun1331 — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +21,8 @@ These changes are available on the `master` branch, but have not yet been releas
 
 - Fixed the type-hinting of `Member.move_to` and `Member.edit` to reflect actual
   behavior. ([#2386](https://github.com/Pycord-Development/pycord/pull/2386))
+- Fixed the type-hinting of `ScheduledEvent.subscribers` to reflect actual behavior.
+  ([#2400](https://github.com/Pycord-Development/pycord/pull/2400))
```
Comment (JustaSqu1d) [inline CHANGELOG.md:25]: Move this to the bottom of the section

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]

File: discord/iterators.py [hunk 3]
```diff
@@ -852,10 +852,10 @@ class ScheduledEventSubscribersIterator(_AsyncIterator[Union["User", "Member"]])
     def __init__(
         self,
         event: ScheduledEvent,
-        limit: int,
+        limit: int | None,
```
Comment (Dorukyum) [inline discord/iterators.py:855]: No default value?

Comment (yoggys) [reply discord/iterators.py:855]: > No default value?

it is only used in `subscribers`:
image
image

**Review by Dorukyum — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**